### DISCUSSION TITLE
Execution of queries except Boolean query

### DISCUSSION THREAD

**Discussion by slisarenko** [Jeroen-G/Explorer#153] [Q&A] [answered]
Recently started learning Elasticsearch and getting to know your library. I started with a [Boolean query,](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html#query-dsl-bool-query) and found in the Elasticsearch documentation such queries as: [Boosting query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-boosting-query.html#query-dsl-boosting-query), [Constant score](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-constant-score-query.html) query and others...
Could you tell me if your library supports such requests or not? Because I think not - I would like to be sure
And is it possible to form the request yourself using your library, something like this:

```
        $query = [
            "query" => [
                "constant_score" => [
                    "filter" => [
                        "term" => [
                            "status" => "active"
                        ]
                    ]
                ]
            ]
        ];

        Cartographer::searh()->selectRaw(json_encode($query))->get();
```
Something similar to what Laravel has: [Raw Methods](https://laravel.com/docs/9.x/queries#raw-methods)

**Comment by Jeroen-G** [comment, marked as answer]
Raw queries are not supported. I like the idea so maybe in the future. However, if you look at the Syntax classes in this package it is very easy to create your own queries. They don't need to be in the package to work, that's how powerful it is.

The constant_score is not yet implemented in the package. You can make it for yourself or even contribute it to the package with a test.

There are many queries in the Elastic docs, so I generally tend to only write those that I personally need myself and let others write the classes that they need.
